File: Development/DisplayStation/frmServicio.vb
Imports POSstation.AccesoDatos

Public Class frmServicio
    Dim EsCancelado As Boolean = False

    Private Sub frmServicio_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim Servicio As String
        Try
            Servicio = My.Settings.NombreServicio

            Try
                'Verifico que esté conectada la llave HASP
                ' POSstation.Fabrica.gasolutionsHasp.IsHasp()

                Me.Controlador.ServiceName = Servicio
                'Hago esta instruccin para saber si el nombre es correcot
                Dim a As Integer = Controlador.Status

                tmrRetardo.Enabled = True
            Catch Ex As POSstation.Fabrica.GasolutionsHaspException
                MessageBox.Show("LLAVE HASP NO ENCONTRADA" & vbNewLine & "Por favor conecte en el puerto USB la llave Gasolutions de verificación de licencia" & vbNewLine & "e intente abrir de nuevo el aplicativo", Application.ExecutablePath, MessageBoxButtons.OK, MessageBoxIcon.Error)
                Me.Close()
            Catch ex As Exception
                Throw New Exception("Problema asignando el servicio " & Servicio & ":" & ex.Message)
            End Try
        Catch ex As Exception
            MsgBox(ex.Message)
            Application.Exit()
        End Try
    End Sub

    Private Sub tmrRetardo_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles tmrRetardo.Tick
        If IsRunning() Then
            tmrRetardo.Enabled = False
            EsCancelado = False
            Me.Hide()

            Dim F As New MainDisplay
            F.ShowDialog()

            Me.Close()
        ElseIf EsCancelado Then
            Me.btnReintentar.Enabled = True
            Me.btnCancelar.Enabled = False
        End If
    End Sub

    Private Function IsRunning() As Boolean
        Me.Controlador.Refresh()
        If Controlador.Status = ServiceProcess.ServiceControllerStatus.Running Then
            Return True
        End If
    End Function

    Private Sub btnCancelar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancelar.Click
        EsCancelado = True
    End Sub

    Private Sub btnReintentar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnReintentar.Click
        Me.tmrRetardo.Enabled = False
        Me.btnCancelar.Enabled = True
    End Sub

   
End Class

File: Development/Controles/frmServicio.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace gasolutions.Configurador
{
    public partial class frmServicio : Form
    {
        Boolean EsCancelado = false;
        string Servicio;

        public frmServicio()
        {
            InitializeComponent();
        }

        public string ServiceName
        { 
          set
          { 
              Servicio = value;
          }
        }
        
        private void btnReintentar_Click(object sender, EventArgs e)
        {
            tmrRetardo.Enabled = false;
            btnCancelar.Enabled = true;
        }

        private void btnCancelar_Click(object sender, EventArgs e)
        {
            EsCancelado = true;
        }

        private void tmrRetardo_Tick(object sender, EventArgs e)
        {
            if (IsRunning())
            {
                //tmrRetardo.Enabled = false;
                //EsCancelado = false;
                //this.Hide();

                //MainForm F = new MainForm();
                //F.ShowDialog();

                //this.Close();
            }
            else
            {
                if (EsCancelado)
                {
                    btnReintentar.Enabled = true;
                    btnCancelar.Enabled = false;
                }
            }
        }

        private Boolean IsRunning()
        {
            Controlador.Refresh();
            if (Controlador.Status == System.ServiceProcess.ServiceControllerStatus.Running)
            {
                return true;
            }
            else
            {
                return false;
            }
        }

        private void frmServicio_Load(object sender, EventArgs e)
        {
            //try
            //{
                

                
            //    //Properties.Settings s = new Properties.Settings();
            //    //Servicio = s.NombreServicio;

            //    try
            //    {

            //        //Verifico que esté conectada la llave HASP
            //        //gasolutions.Factory.gasolutionsHasp.IsHasp();

            //        Controlador.ServiceName = Servicio;
            //        //Hago esta instruccin para saber si el nombre es correcto
            //        Int32 a = (Int32)Controlador.Status;

            //        tmrRetardo.Enabled = true;
            //    }
            //    catch (gasolutions.Factory.GasolutionsHaspException)
            //    {
            //        MessageBox.Show("LLAVE HASP NO ENCONTRADA" + (char)10 + "Por favor conecte en el puerto USB la llave Gasolutions de verificación de licencia" + (char)10 + "e intente abrir de nuevo el aplicativo", Application.ExecutablePath, MessageBoxButtons.OK, MessageBoxIcon.Error);
            //        this.Close();
            //    }
            //    catch (Exception ex)
            //    {
            //        throw new Exception("Problema asignando el servicio " + Servicio + ":" + ex.Message);
            //    }
            //}
            //catch (Exception ex)
            //{
            //    MessageBox.Show(ex.Message);
            //    Application.Exit();
            //}
        }
    }
}